fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/GenericDataServiceImpl.java

```diff
@@ -226,38 +216,16 @@ public List<ResultsetColumnHeaderData> fillResultsetColumnHeaders(final String d
 
             List<ResultsetColumnValueData> columnValues = new ArrayList<>();
             String codeName = null;
-            if ("varchar".equalsIgnoreCase(columnType)) {
-
-                final int codePosition = columnName.indexOf("_cv");
+            final int codePosition = columnName.indexOf("_cd");
+            if ("varchar".equalsIgnoreCase(columnType) || "int".equalsIgnoreCase(columnType) || "integer".equalsIgnoreCase(columnType)) {
                 if (codePosition > 0) {
                     codeName = columnName.substring(0, codePosition);
-
                     columnValues = retreiveColumnValues(codeName);
                 }
-
```

Comment:
Could you please elaborate on these changes? I'm having a hard time understanding them.

Reply:
There were a couple of unused components and I applied some refactoring because there was some duplicated code to do the same steps